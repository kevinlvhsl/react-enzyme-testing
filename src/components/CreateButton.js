import React from 'react'
import Ionicon from 'react-ionicons'

import PropTypes from 'prop-types';
const CreateButton = ({ text, onClickHandler }) => {
  return (
    <button className="btn btn-primary btn-block d-flex justify-content-center align-items-center" onClick={(e) => {onClickHandler(e)}}>
      <Ionicon
        className="rounded-circle"
        fontSize="30px"
        color="#fff"
        icon="ios-add-circle"
        beat={true}
      />
      {text}
    </button>
  )
}
CreateButton.propTypes = {
  text: PropTypes.string,
  onClickHandler: PropTypes.func
}
export default CreateButton
