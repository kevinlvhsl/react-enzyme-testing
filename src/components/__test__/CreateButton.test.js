import React from 'react'
import { shallow } from 'enzyme'
import CreateButton from '../CreateButton'
import Ionicon from 'react-ionicons'
const props = {
  text: 'create',
  onClickHandler: jest.fn()
}
let wrapper
describe('test CreateButton component', () => {
  beforeEach(() => {
    wrapper = shallow(<CreateButton {...props } />)
  })
  it('should render the componentto match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the icon correct', () => {
    expect(wrapper.find(Ionicon).length).toEqual(1)
  })
  it('should click trigger props function', () => {
    wrapper.simulate('click')
    expect(props.onClickHandler).toHaveBeenCalled()
  })

})