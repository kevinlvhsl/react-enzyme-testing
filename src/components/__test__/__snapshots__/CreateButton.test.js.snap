// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CreateButton component should render the componentto match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateButton
    onClickHandler={[MockFunction]}
    text="create"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Ionicon
          beat={true}
          className="rounded-circle"
          color="#fff"
          fontSize="30px"
          icon="ios-add-circle"
          rotate={false}
          shake={false}
          style={Object {}}
        />,
        "create",
      ],
      "className": "btn btn-primary btn-block d-flex justify-content-center align-items-center",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "beat": true,
          "className": "rounded-circle",
          "color": "#fff",
          "fontSize": "30px",
          "icon": "ios-add-circle",
          "rotate": false,
          "shake": false,
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "create",
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Ionicon
            beat={true}
            className="rounded-circle"
            color="#fff"
            fontSize="30px"
            icon="ios-add-circle"
            rotate={false}
            shake={false}
            style={Object {}}
          />,
          "create",
        ],
        "className": "btn btn-primary btn-block d-flex justify-content-center align-items-center",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "beat": true,
            "className": "rounded-circle",
            "color": "#fff",
            "fontSize": "30px",
            "icon": "ios-add-circle",
            "rotate": false,
            "shake": false,
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "create",
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
